core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,51 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class JavaGrepLambdaImp extends JavaGrepImp{
+
+    public static void main(String[] args) {
+        if (args.length != 3){
+            throw new IllegalArgumentException("Usage: JavaGrep regex rootPath outFile");
+        }
+
+        //creating JavaGrepLambdaImp object instead of JavaGrepImp
+        JavaGrepLambdaImp javaGrepImp = new JavaGrepLambdaImp();
+        javaGrepImp.setRegex(args[0]);
+        javaGrepImp.setRootPath(args[1]);
+        javaGrepImp.setOutFile(args[2]);
+
+        try{
+            //calling the parent method, but it will call overridden methods in this class
+            javaGrepImp.process();
+        } catch (Exception ex){
+            ex.printStackTrace();
```

Comment:
you have to use a logger.

Reply:
https://www.notion.so/jarvisdev/Java-Coding-Style-2b03268b4523484d9c505dc36735d00e#7bcaeca874cb473eb0eddf0dd5b171f8